src/Health/HealthController.ts

```diff
@@ -0,0 +1,205 @@
+import BitcoinCore = require('bitcoin-core')
+import { inject, injectable } from 'inversify'
+import { Collection, Db } from 'mongodb'
+import * as Pino from 'pino'
+
+import { childWithFileName } from 'Helpers/Logging'
+import { IPFSHashFailure } from 'Interfaces'
+import { IPFS } from './IPFS'
+
+@injectable()
+export class HealthController {
+  private readonly db: Db
+  private readonly collection: Collection
+  private readonly bitcoinCore: BitcoinCore
+  private readonly logger: Pino.Logger
+  private readonly ipfs: IPFS
+
+  constructor(
+    @inject('Logger') logger: Pino.Logger,
+    @inject('DB') db: Db,
+    @inject('BitcoinCore') bitcoinCore: BitcoinCore,
+    @inject('IPFS') ipfs: IPFS
+  ) {
+    this.logger = childWithFileName(logger, __filename)
+    this.db = db
+    this.collection = this.db.collection('health')
+    this.bitcoinCore = bitcoinCore
+    this.ipfs = ipfs
+  }
+
+  async checkMongoConnection(): Promise<any> {
+    this.logger.trace('MongoHealth', 'Checking Mongo...')
+    const connection = await this.db.stats()
+    const mongoIsConnected = connection.ok === 1 ? true : false
+    await this.collection.updateOne(
```

Comment:
If mongo is not connected, it will never set this to false in mongo. So this work of storing the mongo connection status in the db is not helpful. Can you think of a better way?

Reply:
You're right. If the connection to mongo is broken, we should get error log messages that will trigger alarms, though. Maybe just [confirm this and] remove the information about mongo?

It's the same thing for RMQ, and we should probably follow the same logic for IPFS and Bitcoin Core, though we'd probably need to update the code to check _what_ error is being thrown, due to the retry systems.